tests/forte/image_annotation_test.py

```diff
@@ -30,21 +31,208 @@ class ImageAnnotationTest(unittest.TestCase):
+        region_type = "forte.data.ontology.top.Region"
+        box_type = "forte.data.ontology.top.Box"
+
+        # Analyzing entries in first Data Pack
+        bb_list = list(
+            self.datapack_1._data_store._DataStore__elements[bb_type]
         )
+        bb_entries = list(self.datapack_1._data_store.all_entries(bb_type))
 
-    def test_image_annotation(self):
+        img1_box_list = self.datapack_1.get_image_array(image_payload_idx=0)
+
+        self.assertEqual(bb_list, bb_entries)
         self.assertEqual(
-            self.datapack.image_annotations[0].image_payload_idx, 0
+            self.datapack_1._data_store.num_entries(bb_type), len(bb_list)
+        )
+        self.assertEqual(
+            len(img1_box_list), 6
+        )  # For each bounding box, there is a grid payload created as well
+
+        # Analyzing entries in second Data Pack
+        bb_list = list(
+            self.datapack_2._data_store._DataStore__elements[bb_type]
+        )
+        bb_entries = list(self.datapack_2._data_store.all_entries(bb_type))
+
+        box_list = list(
+            self.datapack_2._data_store._DataStore__elements[box_type]
+        )
+        box_entries = list(self.datapack_2._data_store.all_entries(box_type))
+
+        region_list = list(
+            self.datapack_2._data_store._DataStore__elements[region_type]
+        )
+        region_entries = list(
+            self.datapack_2._data_store.all_entries(region_type)
         )
 
-        self.assertTrue(
-            array_equal(self.datapack.image_annotations[0].image, self.line)
+        # Box and BoundingBox are subclasses of Region
+        self.assertEqual(
+            len(region_list + box_list + bb_list), len(region_entries)
+        )
+        self.assertEqual(
+            self.datapack_2._data_store.num_entries(region_type),
+            len(region_entries),
         )
+
+        # BoundingBox is a subclass of Box
+        self.assertEqual(len(box_list + bb_list), len(box_entries))
+        self.assertEqual(
+            self.datapack_2._data_store.num_entries(box_type), len(box_entries)
+        )
+
+    def test_delete_image_annotations(self):
+
+        box_type = "forte.data.ontology.top.BoundingBox"
+        box_list = len(
+            list(self.datapack_1._data_store._DataStore__elements[box_type])
+        )
+
+        self.datapack_1._data_store.delete_entry(self.bb1.tid)
+        self.assertEqual(
+            self.datapack_1._data_store.num_entries(box_type), box_list - 1
+        )
+
+    def test_update_image_annotation(self):
+        # Check current value
+        self.assertEqual(self.bb1._height, 2)
+
+        # Change a parameter of the entry object
+        self.bb1._height = 5
+
+        # Fetch attribute value from data store
+        bb1_height = self.datapack_1._data_store.get_attribute(
+            self.bb1.tid, "_height"
+        )
+        # Check new value
+        self.assertEqual(bb1_height, 5)
+
```

Comment:
You can also add test cases to update `BoundingBox.image_payload_idx` and see if that change is reflected in `DataStore`